xxh_arm64dispatch.h

```diff
@@ -0,0 +1,86 @@
+/*
+ * xxHash - XXH3 Dispatcher for arm64-based targets
+ * Copyright (C) 2020-2021 Yann Collet
+ * Copyright (C) 2022 Linaro Ltd.
+ *
+ * BSD 2-Clause License (https://www.opensource.org/licenses/bsd-license.php)
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *
+ *    * Redistributions of source code must retain the above copyright
+ *      notice, this list of conditions and the following disclaimer.
+ *    * Redistributions in binary form must reproduce the above
+ *      copyright notice, this list of conditions and the following disclaimer
+ *      in the documentation and/or other materials provided with the
+ *      distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * You can contact the author at:
+ *   - xxHash homepage: https://www.xxhash.com
+ *   - xxHash source repository: https://github.com/Cyan4973/xxHash
+ */
+
+#ifndef XXH_ARM64DISPATCH_H
+#define XXH_ARM64DISPATCH_H
+
+#include "xxhash.h"  /* XXH64_hash_t, XXH3_state_t */
+
+#if defined (__cplusplus)
+extern "C" {
+#endif
+
+XXH64_hash_t  XXH3_64bits_dispatch(const void* input, size_t len);
```

Comment:
This seems to be exactly the same interface contract as `xxh_x86dispatch.c` ?
In which case, maybe it's better to combine them ?

Reply:
There's a bit difference. In `xxh_x86dispatch.h`, the interfaces are defined with XXH_PUBLIC_API. I removed these for `xxh_arm64dispatch.h` since it would result in some build warnings.
```
clang -O3 -march=armv8-a+sve -fPIC -Wall -Wextra -Wstrict-aliasing=1 -std=c99     -DARM64_DISPATCH=1 -I../..     -c -o main.o main.c
In file included from main.c:36:
In file included from ./hashes.h:68:
./../../xxh_arm64dispatch.h:45:30: warning: function 'XXH3_64bits_dispatch' has internal linkage but is not defined [-Wundefined-internal]
XXH_PUBLIC_API XXH64_hash_t  XXH3_64bits_dispatch(const void* input, size_t len);
                             ^
./hashes.h:91:21: note: used here
    return (size_t) XXH3_64bits(src, srcSize);
                    ^
./../../xxh_arm64dispatch.h:65:22: note: expanded from macro 'XXH3_64bits'
# define XXH3_64bits XXH3_64bits_dispatch
                     ^
./../../xxh_arm64dispatch.h:50:30: warning: function 'XXH3_128bits_dispatch' has internal linkage but is not defined [-Wundefined-internal]
XXH_PUBLIC_API XXH128_hash_t XXH3_128bits_dispatch(const void* input, size_t len);
                             ^
./hashes.h:98:21: note: used here
    return (size_t) XXH3_128bits(src, srcSize).low64;
                    ^
./../../xxh_arm64dispatch.h:76:23: note: expanded from macro 'XXH3_128bits'
# define XXH3_128bits XXH3_128bits_dispatch
                      ^
2 warnings generated.
```
I'm not sure whether I could remove these for x86. So I just created a copy with a few differences.

Reply:
It's not clear to me why there are these differences compared to the `x86` dispatcher.
Would be worth an investigation.